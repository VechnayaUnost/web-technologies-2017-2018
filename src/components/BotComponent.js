import React from 'react';
import { connect } from "react-redux";
import ImageComponent from './ImageComponent';

function BotComponent(props) {
    return (
        <div className="BioComponent">
            
            
                
                <ImageComponent icon={props.userSocial}></ImageComponent>       

                
            

                <h6>{props.userSocial}</h6>

                                <ImageComponent icon={props.userCompany}></ImageComponent>       

                <h6>{props.userCompany}</h6>
                <ImageComponent icon={props.userLocation}></ImageComponent>       

                <h6>{props.userLocation}</h6>
                <ImageComponent icon={props.userEmail}></ImageComponent>       

                <h6>{props.userEmail}</h6>
                
        </div>
    );
}

const mapStateToProps = (state) => {
    return {
        userEmail: state.userInfo.email,
        userLocation: state.userInfo.location,
        userSocial: state.userInfo.blog,
        userCompany: state.userInfo.company,
    };
};


export default connect(mapStateToProps)(BotComponent);