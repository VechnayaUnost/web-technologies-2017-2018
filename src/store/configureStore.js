import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

export default (initialState) => {
    const sagaMiddleware = createSagaMiddleware();
    const store = createStore(rootReducer, initialState, applyMiddleware(sagaMiddleware));

    sagaMiddleware.run(rootSaga);
    store.runSaga = () => sagaMiddleware.run(rootSaga);
    store.close = () => store.dispatch(END);

    return store;
};