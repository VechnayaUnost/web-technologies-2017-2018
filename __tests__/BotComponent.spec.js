import React from 'react';
import BotComponent from '../src/components/BotComponent';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('BotComponent component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        userInfo: {
            email: 'neytiri@avatar.com',
            location: 'Pandora Planet',
            blog: '@neyney',
            company: 'AvatarInc'
        }
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('BotComponent component rendered succesfully', () => {
        const botComponent = mount(
            <BotComponent store={store} />
        );
        expect(mountToJson(botComponent)).toMatchSnapshot();
    });

    it('BotComponent component props match state', () => {
        const botComponent = shallow(
            <BotComponent store={store} />
        );
        expect(botComponent.prop('userEmail')).toBe(initialState.userInfo.email);
        expect(botComponent.prop('userLocation')).toBe(initialState.userInfo.location);
        expect(botComponent.prop('userSocial')).toBe(initialState.userInfo.blog);
        expect(botComponent.prop('userCompany')).toBe(initialState.userInfo.company);
    })
});