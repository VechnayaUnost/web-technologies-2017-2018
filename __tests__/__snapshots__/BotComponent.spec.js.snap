// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BotComponent component test BotComponent component rendered succesfully 1`] = `
<Connect(BotComponent)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BotComponent
    dispatch={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userCompany="AvatarInc"
    userEmail="neytiri@avatar.com"
    userLocation="Pandora Planet"
    userSocial="@neyney"
  >
    <div
      className="BioComponent"
    >
      <ImageComponent
        icon="@neyney"
      >
        <div
          className="BioComponent"
        >
          <img
            src="@neyney"
          />
        </div>
      </ImageComponent>
      <h6>
        @neyney
      </h6>
      <ImageComponent
        icon="AvatarInc"
      >
        <div
          className="BioComponent"
        >
          <img
            src="AvatarInc"
          />
        </div>
      </ImageComponent>
      <h6>
        AvatarInc
      </h6>
      <ImageComponent
        icon="Pandora Planet"
      >
        <div
          className="BioComponent"
        >
          <img
            src="Pandora Planet"
          />
        </div>
      </ImageComponent>
      <h6>
        Pandora Planet
      </h6>
      <ImageComponent
        icon="neytiri@avatar.com"
      >
        <div
          className="BioComponent"
        >
          <img
            src="neytiri@avatar.com"
          />
        </div>
      </ImageComponent>
      <h6>
        neytiri@avatar.com
      </h6>
    </div>
  </BotComponent>
</Connect(BotComponent)>
`;
